include/deal.II/base/exceptions.h

```diff
@@ -1541,88 +1541,101 @@ namespace deal_II_exceptions
-          KOKKOS_IF_ON_HOST(({                                               \
-            if (__builtin_expect(!(cond), false))                            \
-              ::dealii::deal_II_exceptions::internals::issue_error_noreturn( \
-                ::dealii::deal_II_exceptions::internals::ExceptionHandling:: \
-                  abort_or_throw_on_exception,                               \
-                __FILE__,                                                    \
-                __LINE__,                                                    \
-                __PRETTY_FUNCTION__,                                         \
-                #cond,                                                       \
-                #exc,                                                        \
-                exc);                                                        \
-          }))                                                                \
-          KOKKOS_IF_ON_DEVICE(({                                             \
-            if (!(cond))                                                     \
-              Kokkos::abort(#cond);                                          \
-          }))                                                                \
-        }
+#      define Assert(cond, exc)                                                \
+        do                                                                     \
+          {                                                                    \
+            KOKKOS_IF_ON_HOST(({                                               \
+              if (__builtin_expect(!(cond), false))                            \
+                ::dealii::deal_II_exceptions::internals::issue_error_noreturn( \
+                  ::dealii::deal_II_exceptions::internals::ExceptionHandling:: \
+                    abort_or_throw_on_exception,                               \
+                  __FILE__,                                                    \
+                  __LINE__,                                                    \
+                  __PRETTY_FUNCTION__,                                         \
+                  #cond,                                                       \
+                  #exc,                                                        \
+                  exc);                                                        \
+            }))                                                                \
+            KOKKOS_IF_ON_DEVICE(({                                             \
+              if (!(cond))                                                     \
+                Kokkos::abort(#cond);                                          \
+            }))                                                                \
+          }                                                                    \
+        while (0)
```

Comment:
This is perhaps nitpicking, but would you be willing to change all of these `while(0)` to `while(false)`? We're in C++ land, after all :-)

Reply:
2f46b793a560477881c44b5f16c9edbb37aa8a91